app/presenters/hyrax/file_set_presenter.rb

```diff
@@ -100,9 +112,15 @@ def link_presenter_class
     def fetch_parent_presenter
```

Comment:
This is looking like we may want a service object; There's a lot of logic packed into this.